### PR TITLE
Fix create directory

### LINKED ISSUE #4594: Create-Directory in tools won't create directory if at least one exists
-  Is this issue blocking (yes/no) no
-  Is this issue causing unreasonable pain (yes/no) no

`Create-Directory` in `tools.ps1`  won't create non-existing directories when given at least one existing path. The reason is that it takes string array, but then does not iterate over it, which due to powershell casting rules will return `$true, $false` in `Test-Path` which casts to `$true` in `if` and skips the folder creation. 

There is no need to check paths when creating foders on the FileSystem provider when using -Force. It wil only create folders that do not exist. 

```powershell
function Create-Directory([string[]] $path) {
  if (!(Test-Path $path)) {
    New-Item -path $path -force -itemType 'Directory' | Out-Null
  }
}

# Pester tests for repro
Describe 'Create-Directory broken' { 
    It 'returns false given at least one non-existing path' { 
        $existingPath = [IO.Path]::GetTempPath()
        $nonExistingPath = "TestDrive:/thisPathDoesNotExist" 

        [string[]] $path = $existingPath, $nonExistingPath
        [bool] $actual = Test-Path $path 

        $actual | Should -BeFalse
    }

    It 'creates folder given one existing path and one non existing path' { 
        $existingPath = [IO.Path]::GetTempPath()
        $nonExistingPath = "TestDrive:/thisPathDoesNotExist" 

        [string[]] $path = $existingPath, $nonExistingPath
        
        Create-Directory $path 

        $nonExistingPath | Should -Exist
    }
}

# fixed
function Create-Directory([string[]] $Path) {
    # no need to check path, New-Item will only create non-existing items on FileSystem provider
    New-Item -Path $path -Force -ItemType 'Directory' | Out-Null
}

Describe 'Create-Directory fixed' { 
    It 'creates folder given one existing path and one non existing path' { 
        $existingPath = [IO.Path]::GetTempPath()
        $nonExistingPath = "TestDrive:/thisPathDoesNotExist" 

        [string[]] $path = $existingPath, $nonExistingPath
        
        Create-Directory $path 

        $nonExistingPath | Should -Exist
    }
}
```

```
Describing Create-Directory broken
  [-] returns false given at least one non-existing path 21ms
    Expected $false, but got $true.
    15:         $actual | Should -BeFalse
  [-] creates folder given one existing path and one non existing path 47ms
    Expected path 'TestDrive:/thisPathDoesNotExist' to exist, but it did not exist.
    26:         $nonExistingPath | Should -Exist

Describing Create-Directory fixed
  [+] creates folder given one existing path and one non existing path 13ms
```

### PR DESCRIPTION
**Pull request by nohwnd**
Closes: #4594

Avoid checking if directory exists, because without looping over all paths, we would not create the directory when at least one of them exists. Instead let the New-Item figure it out for us.

### CODE REVIEW

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -55,10 +55,13 @@ set-strictmode -version 2.0
 $ErrorActionPreference = 'Stop'
 [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
 
-function Create-Directory([string[]] $path) {
-  if (!(Test-Path $path)) {
-    New-Item -path $path -force -itemType 'Directory' | Out-Null
-  }
+function Create-Directory {
+    param(
```
Comment (tmat) [inline eng/common/tools.ps1:59]: Can we keep the simple parameter declaration? I don't see much value in the extra clutter as New-Item would error when the `$path` is not given. Also keep the original casing of `path`.

Comment (nohwnd) [reply eng/common/tools.ps1:59]: Will do.

**Comment by JohnTortugo** [comment]
Ping @nohwnd

**Comment by nohwnd** [comment]
@JohnTortugo what is up? All the PR feedback was addressed. And I cannot merge here. Please :shipit:

**Pull request merged**